src/seedu/addressbook/AddressBook.java

```diff
@@ -305,12 +306,40 @@ private static void setupDefaultFileForStorage() {
     }
 
     /**
-     * Returns true if the given file is acceptable.
-     * The file path is acceptable if it ends in '.txt'
-     * TODO: Implement a more rigorous validity checking.
+     * Returns true if the given file path is acceptable.
+     * The file path is acceptable if it has a folder that exists,
+     * has a file name that is acceptable to the OS,
+     * File name must also contain an extension
      */
     private static boolean isValidFilePath(String filePath) {
-        return filePath.endsWith(".txt");
+        if (filePath == null) {
+            return false;
+        }
+        File filePathToValidate = new File(filePath);
```

Comment:
Oh, I think I should change to `java.nio.file.Path` after looking up that path is generally better.